fnet/data/multichtiffdataset.py

```diff
@@ -0,0 +1,61 @@
+import skimage.external.tifffile as tifffile
+import pandas as pd
+import numpy as np
+import torch
+
+from fnet.data.fnetdataset import FnetDataset
+
+
+class MultiChTiffDataset(FnetDataset):
+    """Dataset for multi-channel tiff files.
+
+    Currently assumes that images are loaded in ZCXY format
+
+    """
+
+    def __init__(self, dataframe: pd.DataFrame = None, path_csv: str = None,
+                 transform_signal=None,
+                 transform_target=None):
+
+        super().__init__(dataframe, path_csv, transform_signal, transform_target)
+
+        self.df['channel_signal'] = [int(ch)
+                                     for ch in self.df['channel_signal']]
+        self.df['channel_target'] = [int(ch)
+                                     for ch in self.df['channel_target']]
+
+        assert all(i in self.df.columns for i in [
+                   'path_tiff', 'channel_signal', 'channel_target'])
+
+    def __getitem__(self, index):
```

Comment:
what's the logic behind has_target?

Reply:
The target can be NaN/None, which implies a dataset item that is not meant for training or evaluation.